otx/algorithms/detection/adapters/mmdet/evaluation/evaluator.py

```diff
@@ -0,0 +1,393 @@
+        if IoU between bounding boxes is greater than 0.
+        Detection mask is resized to detected bounding box size and
+        padded to the same size as ground truth mask in order to compute IoU.
+
+    Returns:
+        np.ndarray: iou between detected masks and ground truth masks
+
+    """
+    det_bboxes, det_masks = det
+    gt_bboxes = gt_masks.get_bboxes()
+    img_h, img_w = gt_masks.height, gt_masks.width
+    ious = bbox_overlaps(det_bboxes, gt_bboxes, mode="iou")
+    if not ious.any():
+        return ious
+    for coord in np.argwhere(ious > 0):
+        m, n = coord
+        det_bbox, det_mask = sanitize_coordinates(det_bboxes[m], img_h, img_w), det_masks[m]
+        gt_bbox, gt_mask = sanitize_coordinates(gt_bboxes[n], img_h, img_w), gt_masks[n]
+        # add padding to det_mask and gt_mask so that they have the same size
+        min_x1 = min(det_bbox[0], gt_bbox[0])
+        min_y1 = min(det_bbox[1], gt_bbox[1])
+        max_x2 = max(det_bbox[2], gt_bbox[2])
+        max_y2 = max(det_bbox[3], gt_bbox[3])
+        det_bbox_h, det_bbox_w = det_bbox[3] - det_bbox[1], det_bbox[2] - det_bbox[0]
+        det_mask = det_mask.resize((det_bbox_h, det_bbox_w))
+        det_mask = det_mask.expand(max_y2 - min_y1, max_x2 - min_x1, det_bbox[1] - min_y1, det_bbox[0] - min_x1)
+        gt_mask = gt_mask.crop(gt_bbox)
+        gt_mask = gt_mask.to_bitmap()
+        gt_mask = gt_mask.expand(max_y2 - min_y1, max_x2 - min_x1, gt_bbox[1] - min_y1, gt_bbox[0] - min_x1)
+        # compute iou between det_mask and gt_mask
+        det_mask = det_mask.to_ndarray()
+        gt_mask = gt_mask.to_ndarray()
+        assert det_mask.shape == gt_mask.shape, f"det_mask.shape={det_mask.shape} != gt_mask.shape={gt_mask.shape}"
+        ious[m, n] = np.sum(det_mask & gt_mask) / np.sum(det_mask | gt_mask)
+    return ious
+
+
+def tpfpmiou_func(  # pylint: disable=too-many-locals
+    det: Tuple[np.ndarray, BitmapMasks], gt_masks: PolygonMasks, cls_scores, iou_thr=0.5
+):
+    """Compute tp, fp, miou for each image.
+
+    Args:
+        det (Tuple[np.ndarray, BitmapMasks]): detected bboxes and masks
+        gt_masks (PolygonMasks): ground truth polygons
+        cls_scores (np.ndarray): class scores
+        iou_thr (float, optional): IoU threshold. Defaults to 0.5.
+
+    Returns:
+        Tuple[np.ndarray, np.ndarray, float]: tp, fp, miou for each image
+    """
+    num_dets = len(det[0])  # M
+    num_gts = len(gt_masks)  # N
+
+    tp = np.zeros(num_dets, dtype=np.float32)  # pylint: disable=invalid-name
+    fp = np.zeros(num_dets, dtype=np.float32)  # pylint: disable=invalid-name
+    gt_covered_iou = np.zeros(num_gts, dtype=np.float32)
+
+    if len(gt_masks) == 0:
+        fp[...] = 1
+        return tp, fp, 0.0
+    if num_dets == 0:
+        return tp, fp, 0.0
+
+    ious = mask_iou(det, gt_masks)  # (M, N)
+
+    assert ious.shape[0] == num_dets, f"{ious.shape[0]} != {num_dets}"
```

Comment:
Please change assert to raise. We have decided to use raise rather than assert.
And Could you explain what is wrong in more detail?